docs/design/aws_under_the_hood.md

```diff
@@ -0,0 +1,302 @@
+more details.
+
+Other documents describe the general architecture of Kubernetes (all nodes run
+Docker; the kubelet agent runs on each node and launches containers; the
+kube-proxy relays traffic between the nodes etc).
+
+By default on AWS:
+
+* Instances run Ubuntu 15.04 (the official AMI).  It includes a sufficiently
+  modern kernel that pairs well with Docker and doesn't require a
+  reboot.  (The default SSH user is `ubuntu` for this and other ubuntu images.)
+* By default we run aufs over ext4 as the filesystem / container storage on the
+  nodes (mostly because this is what GCE uses).
+
+You can override these defaults by passing different environment variables to
+kube-up.
+
+### Storage
+
+AWS supports persistent volumes by using [Elastic Block Store (EBS)](../user-guide/volumes.md#awselasticblockstore).  These can then be
+attached to pods that should store persistent data (e.g. if you're running a
+database).
+
+By default, nodes in AWS use [instance storage](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/InstanceStorage.html)
+unless you create pods with persistent volumes
+[(EBS)](../user-guide/volumes.md#awselasticblockstore).  In general, Kubernetes
+containers do not have persistent storage unless you attach a persistent
+volume, and so nodes on AWS use instance storage.  Instance storage is cheaper,
+often faster, and historically more reliable.  This does mean that you should
+pick an instance type that has sufficient instance storage, unless you can make
+do with whatever space is left on your root partition.
+
+Note: The master uses a persistent volume ([etcd](architecture.md#etcd)) to track
+its state but similar to the nodes, containers are mostly run against instance
+storage, except that we repoint some important data onto the peristent volume.
+
+The default storage driver for Docker images is aufs.  Specifying btrfs (by passing the environment
+variable `DOCKER_STORAGE=btrfs` to kube-up) is also a good choice for a filesystem.  btrfs
+is relatively reliable with Docker and has improved its reliability with modern
+kernels.  It can easily span multiple volumes, which is particularly useful
+when we are using an instance type with multiple ephemeral instance disks.
+
+### AutoScaling
+
+Nodes (but not the master) are run in an
+[AutoScalingGroup](http://docs.aws.amazon.com/AutoScaling/latest/DeveloperGuide/AutoScalingGroup.html)
+on AWS.  Currently auto-scaling (e.g. based on CPU) is not actually enabled
+([#11935](http://issues.k8s.io/11935)).  Instead, the auto-scaling group means
+that AWS will relaunch any nodes that are terminated.
+
+We do not currently run the master in an AutoScalingGroup, but we should
+([#11934](http://issues.k8s.io/11934)).
+
+### Networking
+
+Kubernetes uses an IP-per-pod model.  This means that a node, which runs many
+pods, must have many IPs.  AWS uses virtual private clouds (VPCs) and advanced
+routing support so each pod is assigned a /24 CIDR.  Each pod is assigned a /24
```

Comment:
Lets remove the first part of the sentence (that spans line 114 to 116, since its mentioned in the previous sentence). Please make the following changes:

Delete:
"Each pod is assigned a /24 CIDR;"

Then start the sentence as:
"The assigned CIDR is then......."